src/opencl-platform/cl_init_platform.cpp

```diff
@@ -0,0 +1,374 @@
+#include <iostream>
+#include <limits.h>
+#include <stdlib.h>
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <fstream>
+
+#include "cl_common.h"
+
+cl_uint query_device_type = CL_DEVICE_TYPE_ALL;
+bool cl_is_init = false;
+
+cl_context context;
+cl_command_queue cmd_queue;
+cl_program program;
+
+//////////////////////////////////////
+
+#define KERNELS_SHA256    1
+
+#if    KERNELS_SHA256
+    #include "kernels_sha256.h"
```

Comment:
Why do we need both ways here?

Reply:
We don't, this was just used for debug. Instead of waiting for the previous kernels to compile, I just have them disabled quickly. I can remove these of course.